design-proposals/vm-delete-protection.md

```diff
@@ -0,0 +1,155 @@
+# Overview
+
+VirtualMachine Delete Protection prevents the accidental deletion of VMs from a cluster. As long as deletion protection
+is activated for certain VM objects, they cannot be deleted unless the protection is intentionally removed.
+
+## Motivation
+
+KubeVirt clusters are often managed by automations, CLI commands, 3rd party tools or UI actions, for example to deploy,
+reconfigure or delete VMs. These automations may result in accidental deletion of VMs that should have not been deleted.
+Affected VMs could be critical to application/s the cluster is running, or they may contain crucial data. Therefore, the
+VM deletion protection aims to avoid accidental degradation of applications or loss of data.
+
+## Goals
+
+- Establish a delete protection mechanism for VirtualMachine objects.
```

Comment:
I'd make it similar to non-goals:
```
Create a delete protection for VirtualMachine objects.
```
Because "Establish" could mean to start a new service to have a "mechanism" to prevent deletion (also IMO "mechanism" is an unclear complicated word for it).